.github/workflows/ci-basic-build.yml

```diff
@@ -0,0 +1,51 @@
+# This is a boring build workflow that just builds the project on
+# mac, ubuntu and windows. Just to check that the optional compiler
+# flags that don't work on a developer's computer still build on 
+# other OS. 
+name: Boring Build
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - '**.rs'
+      - '**.toml'
```

Comment:
These path filters are insufficient, and in general I prefer to exclude specific files from affecting general builds (e.g. for doc-only changes).